src/pages/Admin/Users/Add/index.jsx

```diff
@@ -0,0 +1,353 @@
+ version 2 as published by the Free Software Foundation.
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License along
+ with this program; if not, write to the Free Software Foundation, Inc.,
+ 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+import React, { useEffect, useState } from "react";
+import Title from "components/Title";
+
+// widgets
+import { Alert, Button, InputContainer, Spinner } from "components/Widgets";
+
+// services
+import { getAllFolders } from "services/folders";
+
+// predefined common fields components
+import CommonFields from "components/Upload/CommonFields";
+
+// constants
+import { accessLevels } from "constants/constants";
+
+const AddUser = () => {
+  // initial state for user data
+  const initialAddUserData = {
+    username: "",
+    description: "",
+    email: "",
+    permission: 1,
+    folder: 1,
+    pass1: "",
+    pass2: "",
+    enote: "y",
+    accessLevel: "public",
+    Check_agent_copyright: 0,
+    Check_agent_ecc: 0,
+    Check_agent_keyword: 0,
+    Check_agent_mimetype: 0,
+    Check_agent_monk: 0,
+    Check_agent_nomos: 0,
+    Check_agent_ojo: 0,
+    Check_agent_pkgagent: 0,
+    Check_agent_reso: 0,
+    Check_agent_shagent: 0,
+    default_bucketpool_fk: 2,
+  };
+
+  // initial state to keep a track of the checkbox values
+  const [checkboxes, setCheckboxes] = useState({
+    enote: true,
+    Check_agent_copyright: false,
+    Check_agent_ecc: false,
+    Check_agent_keyword: false,
+    Check_agent_mimetype: false,
+    Check_agent_monk: false,
+    Check_agent_nomos: false,
+    Check_agent_ojo: false,
+    Check_agent_pkgagent: false,
+    Check_agent_reso: false,
+    Check_agent_shagent: false,
+  });
+
+  const bucketPool = [
+    {
+      id: 0,
+      name: "GPL Demo bucket pool, v1",
+      disabled: false,
+    },
+  ];
+
+  const initialMessage = {
+    type: "success",
+    text: "",
+  };
+
+  // initializing all the local states
+  const [addUserData, setAddUserData] = useState(initialAddUserData);
+  const [folderlist, setFolderlist] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [message, setMessage] = useState(initialMessage);
+  const [showMessage, setShowMessage] = useState(false);
+
+  // API call to add user
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    // preparing the final user data to send to the server
+    const finalUserData = { ...addUserData, public: addUserData.accessLevel };
+
+    delete finalUserData.accessLevel;
+    finalUserData.Check_agent_copyright = checkboxes.Check_agent_copyright
+      ? 1
+      : 0;
+    finalUserData.Check_agent_ecc = checkboxes.Check_agent_ecc ? 1 : 0;
```

Comment:
Any specific reasons to use `1 & 0` instead of `true & false`?

Reply:
But the REST API gonna accept true/false so you don't need to worry about this translation.

P.s. Hence you design API first ;-)

Reply:
Changed to `true & false` instead of `0 & 1`.